sdk/src/record-scanner.ts

```diff
@@ -0,0 +1,302 @@
+                    body: JSON.stringify(filter),
+                }),
+            );
+
+            return await response.json();
+        } catch (error) {
+            console.error(`Failed to get owned records: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Find a credits record in the record scanner service.
+     * 
+     * @param {number} microcredits The amount of microcredits to find
+     * @param {OwnedFilter} searchParameters The filter to use to find the record
+     * @returns {Promise<OwnedRecord>} The record
+     */
+    async findCreditsRecord(microcredits: number, searchParameters: OwnedFilter): Promise<OwnedRecord> {
+        try {
+            const records = await this.findRecords({
+                ...searchParameters,
+                program: "credits.aleo",
+                record: "credits",
+                decrypt: true,
+            });
+
+            const record = records.find(record => {
+                const plaintext = RecordPlaintext.fromString(record.recordPlaintext);
+                const amountStr = plaintext.getMember("microcredits").toString();
+                const amount = parseInt(amountStr.replace("u64", ""));
+                return amount >= microcredits;
+            });
+
+            if (!record) {
+                throw new Error("Record not found");
+            }
+
+            return record;
+        } catch (error) {
+            console.error(`Failed to find credits record: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Find credits records in the record scanner service.
+     * 
+     * @param {number[]} microcreditAmounts The amounts of microcredits to find
+     * @param {OwnedFilter} searchParameters The filter to use to find the records
+     * @returns {Promise<OwnedRecord[]>} The records
+     */
+    async findCreditsRecords(microcreditAmounts: number[], searchParameters: OwnedFilter): Promise<OwnedRecord[]> {
+        try {
+            const records = await this.findRecords({
+                ...searchParameters,
+                program: "credits.aleo",
+                record: "credits",
+                decrypt: true,
+            });
+            return records.filter(record => {
+                const plaintext = RecordPlaintext.fromString(record.recordPlaintext);
+                const amount = plaintext.getMember("microcredits").toString();
+                return microcreditAmounts.includes(parseInt(amount.replace("u64", "")));
+            });
+        } catch (error) {
+            console.error(`Failed to find credits records: ${error}`);
+            throw error;
+        }
+    }
+
+    /**
+     * Wrapper function to make a request to the record scanner service and handle any errors
+     * 
+     * @param {Request} req The request to make
+     * @returns {Promise<Response>} The response
+     */
+    private async recordScannerServiceRequest(req: Request): Promise<Response> {
```

Comment:
```suggestion
    private async request(req: Request): Promise<Response> {
```

Can we just change this method to request?

Reply:
done